Fix for menu in dark mode (2nd try)

uuf6429 opened the pull request:
Adding CSS filters causes a change in page containers, in this case causing the (usually) sticky menu to become static.

More information about this: https://stackoverflow.com/a/52937920/314056

In principle, using filters has some limits and/or side effects. This could be such a case. I've managed to find a workable solution in this PR, (that I tested a bit more than the last), but it's not exactly pretty. :)

uuf6429 committed:
Avoid filter from interfering with page structure

uuf6429 committed:
Fix menu without affecting background

ondrejmirtes commented:
No merge commits please. Clean linear history on top of 1.11.x. Thank you very much!

ondrejmirtes commented:
Alright, can you explain in your own words why this new version works? 😊

uuf6429 commented:
Sure - the sticky menu uses `position: fixed`. All types of positioning depend on the parent. When we set a filter on the body element, suddenly it becomes "the container", so fixed position becomes relatively to the body and not the entire viewport (I guess html?).

The fix works by targeting other elements as top-level as possible (we want that for global dark mode) but not too top (to avoid messing up positioning).

In other words, this problem we just solved will come up again if you have other `position: fixed` elements inside the page (anywhere under the filter level).

The white background from last time was just a situation I forgot to anticipate, it is however unrelated.

uuf6429 commented:
Oh, and the top-level `div` has a white(ish) background by default, since the filter is one level below it now, it had to be changed to black.

I could make the second selector more specific since there's only a `nav` and a `div` on the 2nd level.

uuf6429 committed:
Use exact inverse bg

ondrejmirtes commented:
Awesome, thank you! Tested it locally and could not find any glitch with this solution :)

BTW this dark mode project is a perfect example of "this will just take 5 minutes!" And the initial version did. But the version that actually works and looks nice took a few more weeks 😂 

I had a similar experience with migration from homegrown ORM to Doctrine at my last job. My colleague said "It's easy, will take just a single morning." Sure, it did just take a single morning. And then 10 months of top of that 😂
